int8 left
int8 right
---
string message
